### ISSUE TITLE
tfe_workspace's organization parameter is case sensitive but API is not

### ISSUE THREAD

**Issue by dekimsey** [hashicorp/terraform-provider-tfe#299]
The TFE workspace organization parameter in the provider is case-sensitive, but the TFE API is not.

It is stored and represented as `MyCompany` in the UI, and the provider handles this gracefully. But the TFE API appears to not care about case itself, and our terraform remote backends are all lower cased and work fine. Only the TFE provider seems to be particular about the case.

This creates inconsistencies in implementation/usage.

With the curl API:
1. Create `MyCompany`, a case-sensitive organization in TFE.
2. curl https://app.terraform.io/api/v2/organizations/MyCompany
3. curl https://app.terraform.io/api/v2/organizations/mycompany
4. curl https://app.terraform.io/api/v2/organizations/MYCOMPANY

Using Terraform this creates odd behaviors:

Given:
```
terraform {
  required_version = ">= 0.14"
  backend "remote" {
    organization = "mycompany"
    workspaces {
      name = "demo"
    }
  }
}
resource "tfe_workspace" "this" {
   name                  = "case-test"
   organization       = "mycompany"
}
1. `terraform init`, works, remote doesn't care about casing
2. `terraform apply`, works, workspace is created.
3. `terraform apply`, fail, workspace is replaced even though API is case-insensitive.
```

**Comment by bendrucker** [comment]
This is easy enough to fix with the Terraform provider SDK, opened #300

**Comment by bendrucker** [comment]
Forgot to put `closes #299` in #300, this can be closed

**Comment by radditude** [comment]
Thanks!
